blockchain/blockchain.go

```diff
@@ -1045,23 +1059,51 @@ func (bc *BlockChain) writeStateTrie(block *types.Block, state *state.StateDB) e
 			trieDB.Commit(block.Header().Root, true, block.NumberU64())
 		}
 
-		if current := block.NumberU64(); current > triesInMemory {
-			// Find the next state trie we need to commit
-			header := bc.GetHeaderByNumber(current - triesInMemory)
-			chosen := header.Number.Uint64()
+		// trigger to GC stateDB cache
+		select {
+		case bc.gcTrigger <- block.NumberU64():
+		default:
```

Comment:
If block number is not able to be passed to the `bc.gcTrigger` channel, it does nothing. Is it what you expected?

Reply:
Yes it is. This routine is main routine. 
So if this trigger can't pass the signal and it waits, the main routine has some problem. 

I don't want this situation. Therefore this signal is not buffered. 
But, if a trigger is ignored, it will try again in next block.

Reply:
Hmm... but you need to take care of this because it may change original behavior. Please add some logs for the unwanted case.

Reply:
I also agree with the @kjhman21 's point that we need to leave a log here. 
Log would be needed for the both cases.

Reply:
I added a log here and inside of the loop. Thanks.

Reply:
What about adding a log for `default:` case? I think this is unusual and in that case one block number can be missing. Can we use `logger.Error()` for `default:` case?

Reply:
That default situation is not error case. 
It is intended case.
So How about `logger.Warn()`?

Reply:
logger.Warn() seems to be enough for this case (unless it skips too much)

Reply:
I added warn log. Thanks. :)